server/src/main/java/org/apache/druid/segment/indexing/TuningConfig.java

```diff
@@ -37,6 +37,7 @@
   int DEFAULT_MAX_PARSE_EXCEPTIONS = Integer.MAX_VALUE;
   int DEFAULT_MAX_SAVED_PARSE_EXCEPTIONS = 0;
   int DEFAULT_MAX_ROWS_IN_MEMORY = 1_000_000;
+  int DEFAULT_MAX_MEMORY_PERCENTAGE_OF_JVM = 16;
```

Comment:
I think `DEFAULT_MAX_MEMORY_PERCENTAGE_OF_JVM` should be a parameter of the appendable index.
A different index, e.g., off-heap, can use up to half the off-heap memory by default for example.
This can be implemented in `AppendableIndexSpec` as `getDefaultMaxMemoryPercentage()`.